wgpu/src/context.rs

```diff
@@ -0,0 +1,1873 @@
+use std::{any::Any, fmt::Debug, future::Future, ops::Range, pin::Pin};
+
+use wgt::{
+    AdapterInfo, BufferAddress, BufferSize, Color, CompositeAlphaMode, DownlevelCapabilities,
+    DynamicOffset, Extent3d, Features, ImageDataLayout, ImageSubresourceRange, IndexFormat, Limits,
+    PresentMode, ShaderStages, SurfaceConfiguration, SurfaceStatus, TextureFormat,
+    TextureFormatFeatures,
+};
+
+use crate::{
+    backend::BufferMappedRange, BindGroupDescriptor, BindGroupLayoutDescriptor, Buffer,
+    BufferAsyncError, BufferDescriptor, CommandEncoderDescriptor, ComputePassDescriptor,
+    ComputePassInner, ComputePipelineDescriptor, Context, DeviceDescriptor, Error, ErrorFilter,
+    ImageCopyBuffer, ImageCopyTexture, Maintain, MapMode, PipelineLayoutDescriptor,
+    QuerySetDescriptor, RenderBundleDescriptor, RenderBundleEncoderDescriptor, RenderInner,
+    RenderPassDescriptor, RenderPassInner, RenderPipelineDescriptor, RequestAdapterOptions,
+    RequestDeviceError, SamplerDescriptor, ShaderModuleDescriptor, ShaderModuleDescriptorSpirV,
+    Texture, TextureDescriptor, TextureViewDescriptor, UncapturedErrorHandler,
+};
+
+pub struct IdSendSync {
+    inner: Box<dyn Any + Send + Sync>,
```

Comment:
Is there any way we can avoid these boxes, this is quite a few extra indirections which would be great to avoid.